### PR TITLE
Pubsub func testing

### PR DESCRIPTION
**Pull request by ludwikbukowski**
Fixed rebar3 integration and publish_entry_children function

### CODE REVIEW

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'meck' is not using appropriate protocol, please change this to something like 'https://.*'

File: src/escalus_pubsub_stanza.erl [hunk 1]
```diff
@@ -171,6 +176,14 @@ create_publish_node_content_stanza_third(NodeName, ItemId) ->
...
+create_publish_node_content_stanza_with_timestamp(NodeName, ItemId) ->
+    PublishEntry = publish_entry(entry_body_with_timestamp()),
+    ItemTopublish = publish_item(ItemId, PublishEntry),
+    PublNode = publish_node_with_content_stanza(NodeName, ItemTopublish),
+    pubsub_stanza([PublNode], ?NS_PUBSUB).
+
+
 retract_from_node_stanza(NodeName, ItemId) ->
     ItemToRetract = #xmlel{name = <<"item">>, attrs=[{<<"id">>, ItemId}], children=[]},
     RetractNode =  #xmlel{name = <<"retract">>, attrs=[{<<"node">>, NodeName}], children=[ItemToRetract]},
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 189 is too long:     RetractNode =  #xmlel{name = <<"retract">>, attrs=[{<<"node">>, NodeName}], children=[ItemToRetract]},.

File: src/escalus_pubsub_stanza.erl [hunk 2]
```diff
@@ -85,7 +90,13 @@ iq_set_get_rest(SrcIq, Id, From) ->
 entry_body_sample1() ->
     [
      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 92 is too long:      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},.

File: src/escalus_pubsub_stanza.erl [hunk 3]
```diff
@@ -50,6 +51,8 @@ pubsub_stanza(Children, NS) ->
            attrs = [{<<"xmlns">>, NS} ],
            children = Children  }.
 
+
+
 publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItemId, User, SampleNumber) ->
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 56 is too long: publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItemId, User, SampleNumber) ->.

File: src/escalus_pubsub_stanza.erl [hunk 4]
```diff
@@ -58,6 +61,8 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
                             create_publish_node_content_stanza_second(DestinationTopicName, PublishItemId);
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 61 is too long:                             create_publish_node_content_stanza_second(DestinationTopicName, PublishItemId);.

File: src/escalus_pubsub_stanza.erl [hunk 5]
```diff
@@ -58,6 +61,8 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
...
                         sample_three ->
                             create_publish_node_content_stanza_third(DestinationTopicName, PublishItemId);
+                        sample_time ->
+                            create_publish_node_content_stanza_with_timestamp(DestinationTopicName, PublishItemId);
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 65 is too long:                             create_publish_node_content_stanza_with_timestamp(DestinationTopicName, PublishItemId);.

File: src/escalus_pubsub_stanza.erl [hunk 6]
```diff
@@ -58,6 +61,8 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
                             create_publish_node_content_stanza_second(DestinationTopicName, PublishItemId);
                         sample_three ->
                             create_publish_node_content_stanza_third(DestinationTopicName, PublishItemId);
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 63 is too long:                             create_publish_node_content_stanza_third(DestinationTopicName, PublishItemId);.

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'wsecli' is not using appropriate protocol, please change this to something like 'https://.*'

File: src/escalus_pubsub_stanza.erl [hunk 7]
```diff
@@ -85,7 +90,13 @@ iq_set_get_rest(SrcIq, Id, From) ->
 entry_body_sample1() ->
     [
      #xmlel{name = <<"title">>, children  = [ #xmlcdata{content=[<<"The title of content.">>]}]},
-     #xmlel{name = <<"summary">>, children= [ #xmlcdata{content=[<<"To be or not to be...">>]}]}
+     #xmlel{name = <<"summary">>, children = [ #xmlcdata{content=[<<"To be or not to be...">>]}]}
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 93 is too long:      #xmlel{name = <<"summary">>, children = [ #xmlcdata{content=[<<"To be or not to be...">>]}]}.

File: src/escalus_pubsub_stanza.erl [hunk 8]
```diff
@@ -85,7 +90,13 @@ iq_set_get_rest(SrcIq, Id, From) ->
...
+    ].
+
+entry_body_with_timestamp() ->
+    MicroSec = usec:from_now(os:timestamp()),
+    [
+     #xmlel{name = <<"MSG_SENT_AT">>, children  = [ #xmlcdata{content=[integer_to_binary(MicroSec)]}]}
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 99 is too long:      #xmlel{name = <<"MSG_SENT_AT">>, children  = [ #xmlcdata{content=[integer_to_binary(MicroSec)]}]}.

File: src/escalus_pubsub_stanza.erl [hunk 9]
```diff
@@ -58,6 +61,8 @@ publish_sample_content_stanza(DestinationTopicName, DestinationNode, PublishItem
                             create_publish_node_content_stanza_second(DestinationTopicName, PublishItemId);
                         sample_three ->
                             create_publish_node_content_stanza_third(DestinationTopicName, PublishItemId);
+                        sample_time ->
+                            create_publish_node_content_stanza_with_timestamp(DestinationTopicName, PublishItemId);
                         _ ->
                             create_publish_node_content_stanza(DestinationTopicName, PublishItemId)
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 67 is too long:                             create_publish_node_content_stanza(DestinationTopicName, PublishItemId).

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'usec' revision is specified 'master', please change this to a tag, branch or specific commit.

File: src/escalus_pubsub_stanza.erl [hunk 10]
```diff
@@ -171,6 +176,14 @@ create_publish_node_content_stanza_third(NodeName, ItemId) ->
...
+%% timestamp is microseconds
+create_publish_node_content_stanza_with_timestamp(NodeName, ItemId) ->
+    PublishEntry = publish_entry(entry_body_with_timestamp()),
+    ItemTopublish = publish_item(ItemId, PublishEntry),
+    PublNode = publish_node_with_content_stanza(NodeName, ItemTopublish),
+    pubsub_stanza([PublNode], ?NS_PUBSUB).
+
+
 retract_from_node_stanza(NodeName, ItemId) ->
     ItemToRetract = #xmlel{name = <<"item">>, attrs=[{<<"id">>, ItemId}], children=[]},
```
Comment (gadgetci) [inline src/escalus_pubsub_stanza.erl]: According to **Elvis**:
> Line 188 is too long:     ItemToRetract = #xmlel{name = <<"item">>, attrs=[{<<"id">>, ItemId}], children=[]},.

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'fusco' is not using appropriate protocol, please change this to something like 'https://.*'

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'usec' is not using appropriate protocol, please change this to something like 'https://.*'

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'base16' is not using appropriate protocol, please change this to something like 'https://.*'

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'exml' is not using appropriate protocol, please change this to something like 'https://.*'

**Comment by gadgetci** [comment]
According to **Elvis**:
> Dependency 'mustache' is not using appropriate protocol, please change this to something like 'https://.*'

**Comment by lavrin** [comment]
Looks good to me. Let's see what @astachurski says when he's back.

**Pull request merged**